src/bin/pg_dump/pg_dump.c

```diff
@@ -11371,6 +11382,10 @@ dumpDomain(Archive *fout, const TypeInfo *tyinfo)
 
 	appendPQExpBufferStr(q, ";\n");
 
+	if (typsupporttypmod)
+		appendPQExpBuffer(q,
+						"SET enable_domain_typmod = FALSE;\n");
```

Comment:
Let's restore the previous value instead of putting a hard-coded value.

Reply:
Fixed!